main.go

```diff
@@ -8,64 +8,60 @@ import (
 )
 
 // validateIPv4 function to check a given string is valid ipv4 or not.
-func validateIPv4(line string) bool {
+func validateIPv4(line string) (bool, error) {
 	ipv4, err := regexp.Compile(`^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$`)
 	if err != nil {
-		log.Fatal(err)
+		return false, err
 	}
 	if ipv4.MatchString(line) {
-		return true
+		return true, err
 	} else {
-		return false
+		return false, err
 	}
```

Comment:
**Suggestion:** The validateIPv4 and validateIPv6 functions are using the same logic to validate the IP addresses. You can refactor these two functions into a single function that takes the regular expression as an argument. This will reduce code duplication and make the code more maintainable.
```suggestion
func validateIP(line string, pattern string) (bool, error) {
	ip, err := regexp.Compile(pattern)
	if err != nil {
		return false, err
	}
	if ip.MatchString(line) {
		return true, nil
	} else {
		return false, nil
	}
}
```